### ISSUE TITLE
[Feature Request] Additional Event Hub Properties for the Management Library

### ISSUE THREAD

**Issue by srivathsanvlb** [Azure/azure-sdk-for-net#10014]
**Query/Question**
When using the WindowsAzure.ServiceBus nuget package, in the Microsoft.ServiceBus namespace we have a class NamespaceManager. It has some methods like GetEventHub and GetEventHubPartition. It helps to retrieve many properties for existing Event Hubs. In .NET core, I think the equivalent is Microsoft.Azure.Management.EventHub & Azure.Messaging.EventHubs namespace. But some of the Event Hub properties that can be retrieved in  .NET Framework are not retrievable in .net core. 

  | **.NET **Framework**** |   | **.NET Core**
-- | -- | -- | --
  |   |   |  
**Class Name** | **Property Name** | **Class Name** | **Property Name**
PartitionDescription | PartitionId |   |  
PartitionDescription | EventHubPath |   |  
PartitionDescription | ConsumerGroupName |   |  
PartitionDescription | EndSequenceNumber |   |  
PartitionDescription | BeginSequenceNumber | PartitionProperties | BeginningSequenceNumber
PartitionDescription | **SizeInBytes** |   |  
PartitionDescription | **IncomingBytesPerSecond** |   |  
PartitionDescription | **OutgoingBytesPerSecond** |   |  
PartitionDescription | LastEnqueuedOffset | PartitionProperties | LastEnqueuedOffset
PartitionDescription | LastEnqueuedTimeUtc | PartitionProperties | LastEnqueuedTime
  |   |   |  
EventHubDescription | Path |   |  
EventHubDescription | MessageRetentionInDays | Eventhub | MessageRetentionInDays
EventHubDescription | Authorization |   |  
EventHubDescription | **Status** |   |  
EventHubDescription | CreatedAt | Eventhub | CreatedAt
EventHubDescription | UpdatedAt | Eventhub | UpdatedAt
EventHubDescription | UserMetadata |   |  
EventHubDescription | PartitionCount | Eventhub | PartitionCount
EventHubDescription | PartitionIds | Eventhub | PartitionIds

The ones in bold are the ones I am specifically looking for.

**Environment:**
 - Microsoft.Azure.Management.EventHub 2.5.0
 - .NET Core SDK (reflecting any global.json):
 Version:   3.1.101
 Commit:    b377529961

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.101\

Host (useful for support):
  Version: 3.1.1
  Commit:  a1388f194c

.NET Core SDKs installed:
  3.1.101 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

 - IDE and version : Visual Studio 16.4.5

**Comment by wantedfast** [comment]
Hi @srivathsanvlb 

Packages [WindowsAzure.ServiceBus](https://www.nuget.org/packages/WindowsAzure.ServiceBus), [Microsoft.Azure.Management.EventHub](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Microsoft.Azure.Management.EventHub), and [Azure.Messaging.EventHubs](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs) are different packages. Azure.Messaging.EventHubs is new SDK for EventHub which have more unit design compared to Microsoft.Azure.Management.EventHub.  Actually, it is hard to say that WindowsAzure.ServiceBus and Microsoft.Azure.Management.EventHub are equivalents even both of them can be used to operate with EventHubs. All 3 packages support both .NET Framework and .NET Core.  
If just switch .NET Framework to .NET Core, **please continue to use WindowsAzure.ServiceBus in your .NET Core project directly**. Of course, if you want to refactor the Application, the latest EventHub package Azure.Messaging.EventHubs is strongly recommended to be the package to interact with EventHub.

@jongio

**Comment by srivathsanvlb** [comment]
Ok, Actually I am trying to use it in a .NET Standard project. But I am facing the following issue. 

![image](https://user-images.githubusercontent.com/35536827/74932125-10b4a580-5407-11ea-85e6-633ea21ed642.png)

I think they are talking about the same in https://github.com/dotnet/wcf/issues/3172

Please let me know on how to proceed.

**Comment by serkantkaraca** [comment]
The failure is due to fact that MicrosoftAzure.ServiceBus SDK doesn't support .NET Standard.

**Comment by srivathsanvlb** [comment]
Thanks @serkantkaraca . Actually I need the above said properties in a .NET standard project.
Can any one let me know where I can get them?

**Comment by jongio** [comment]
@serkantkaraca - Do you have that info for @srivathsanvlb?

**Comment by serkantkaraca** [comment]
Sorry for late response.

There are 2 APIs you can use to access them.

https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Microsoft.Azure.EventHubs/4.1.1/api/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.EventHubClient.html#Microsoft_Azure_EventHubs_EventHubClient_GetRuntimeInformationAsync

https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Microsoft.Azure.EventHubs/4.1.1/api/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.EventHubClient.html#Microsoft_Azure_EventHubs_EventHubClient_GetPartitionRuntimeInformationAsync_System_String_

See the returned object descriptions on the API page. Let me know if you have any questions.

**Comment by srivathsanvlb** [comment]
Hi @serkantkaraca 

Thanks for your reply. But I don't see the properties that I am looking for. 

To re-iterate, I am specifically looking for the following properties that are available in .NET Framework. Methods like GetEventHub and GetEventHubPartition are used for getting the below property values. I am looking for the equivalent in .NET Standard. Kindly let me know about the same.


  | .NET Framework |   | .NET Standard
-- | -- | -- | --
  |   |   |  
**Class Name** | **Property Name** | 
PartitionDescription | SizeInBytes |   |  
PartitionDescription | IncomingBytesPerSecond |   |  
PartitionDescription | OutgoingBytesPerSecond |   |  
EventHubDescription | Status |   |

**Comment by serkantkaraca** [comment]
Unfortunately, those are not available from .Net Standard client.

**Comment by srivathsanvlb** [comment]
@serkantkaraca oh ok

We are currently working to upgrade .NET framework projects to .NET Standard/.NET Core.
We are facing some blockers similar to the above. What is the right way to ask Microsoft to consider our request for enabling .NET Core/Standard to expose properties/methods that we use in .NET Framework?

**Comment by serkantkaraca** [comment]
You should consider upgrading to https://www.nuget.org/packages/Azure.Messaging.EventHubs/ instead. We are not adding any new features to Microsoft.Azure.EventHubs.

For any missing feature, create a new issue so SDK Team can track them.

**Comment by jsquire** [comment]
My understanding after looking at #10421 and speaking with @serkant is that the requested properties are only available within the [management REST API](https://docs.microsoft.com/en-us/rest/api/eventhub/get-partition-runtime-metadata) and not currently exposed on the data plane.   I'll ask Serkant to please correct me if I misunderstood that conversation.

For the client libraries, we'll continue the discussion on the potential to include them using #10421, but I believe that there is merit to considering this as a gap in the management library as well.   In the short term, addressing via including in the management library is likely to be the most expedient path to unblocking @srivathsanvlb.

//cc: @wantedfast, @jongio

**Comment by srivathsanvlb** [comment]
@serkantkaraca  @jsquire 

I will use the REST API as of now. Thanks for the quick turn around. You may have this thread open or close it. I will leave it to your discretion.

**Comment by ArunPrabhuM** [comment]
@jsquire , Is there a similar API available for managing the Relays using Shared access signature. I don't find it in the documentation.

**Comment by jsquire** [comment]
> @jsquire , Is there a similar API available for managing the Relays using Shared access signature. I don't find it in the documentation.

On that, I'll have to defer to @serkantkaraca, as beyond the [documentation](https://docs.microsoft.com/en-us/rest/api/relay/), I don't have any insight to offer.

**Comment by jsquire** [comment]
#10421 has been closed after confirming that the requested information is available only within the management plane.  The properties under discussion are currently not exposed within the management library and require calling the REST API directly to retrieve.   I believe that we should consider added them to the management library.

**Comment by serkantkaraca** [comment]
> @jsquire , Is there a similar API available for managing the Relays using Shared access signature. I don't find it in the documentation.

You mean REST API? If so, you can find them here - https://docs.microsoft.com/en-us/rest/api/relay/

**Comment by srivathsanvlb** [comment]
@serkantkaraca 

> > @jsquire , Is there a similar API available for managing the Relays using Shared access signature. I don't find it in the documentation.
> 
> You mean REST API? If so, you can find them here - https://docs.microsoft.com/en-us/rest/api/relay/

These REST APIs must be authenticated using a bearer token ie oauth authentication. I think @ArunPrabhuM  is looking for something similar to https://docs.microsoft.com/en-us/rest/api/eventhub/get-partition-runtime-metadata where the REST API can be authenticated using a Shared Access Signature instead of Bearer OAuth tokens.

**Comment by serkantkaraca** [comment]
Yes, both Wcf Relays and HybridConnections support management over HTTP with SAS. What APIs are you looking for?

**Comment by srivathsanvlb** [comment]
https://docs.microsoft.com/en-us/rest/api/relay/wcfrelays/get
@serkantkaraca 
We are trying to use the above REST API endpoint with SAS authentication

**Comment by isra-fel** [comment]
@jfggdl Hi, this is a feature request to the management plane SDK of Event Hub to expose this[ REST API ](https://docs.microsoft.com/en-us/rest/api/eventhub/get-partition-runtime-metadata). Could you consider and prioritize  it? Thank you.

**Comment by samuelkoppes** [comment]
We have accepted this ask into our internal backlog for evaluation for prioritization in the first half of 2021. Thank you for sharing!

**Comment by ramya-rao-a** [comment]
@kasun04 Any updates here?

**Comment by kasun04** [comment]
We are evaluating the feasibility of adding this feature to the management plane SDK. Will get back to you once we have an update.

**Comment by navba-MSFT** [comment]
@srivathsanvlb Apologies for the late reply. Thanks for reaching out to us. We are looking into this ask. We will update this thread once we have more details on this. 

cc @kasun04
